src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -333,8 +333,12 @@ static void report_broker_config(
       "log.retention.ms",
       config::shard_local_cfg().delete_retention_ms,
       include_synonyms,
-      [](const std::chrono::milliseconds& ms) {
-          return ssx::sformat("{}", ms.count());
+      [](const std::optional<std::chrono::milliseconds>& ret) {
+          if (!ret) {
+              return ssx::sformat("{}", -1);
+          } else {
+              return ssx::sformat("{}", ret->count());
+          }
```

Comment:
@jcsp aside: This is unfortunate, I'd have expected it to use `void property::print(std::ostream& o)` by default, and for the override to take the `const property&` rather than the result of `property::value`.

Reply:
nit: it could also be implemented as:
```suggestion
          return ssx::sformat(
            "{}", ret.value_or(std::chrono::milliseconds{-1}).count());
```

Reply:
@BenPope I think I'm missing context-- is this something I changed recently?

Reply:
Sorry @jcsp I should have used blame, instead of guessing. Not your doing!